src/hooks/__tests__/useTxPendingStatus.ts

```diff
@@ -0,0 +1,105 @@
+import type { JsonRpcProvider } from 'ethers'
+import * as useChainIdHook from '@/hooks/useChainId'
+import { useTxMonitor } from '@/hooks/useTxPendingStatuses'
+import * as web3 from '@/hooks/wallets/web3'
+import * as txMonitor from '@/services/tx/txMonitor'
+import { PendingStatus, type PendingTxsState, PendingTxType } from '@/store/pendingTxsSlice'
+import { renderHook } from '@/tests/test-utils'
+
+describe('useTxMonitor', () => {
+  let mockProvider
+  beforeEach(() => {
+    jest.clearAllMocks()
+
+    jest.spyOn(useChainIdHook, 'default').mockReturnValue('11155111')
+
+    mockProvider = jest.fn() as unknown as JsonRpcProvider
+    jest.spyOn(web3, 'useWeb3ReadOnly').mockReturnValue(mockProvider)
+  })
+
+  it('should not monitor transactions if provider is not available', () => {
+    jest.spyOn(web3, 'useWeb3ReadOnly').mockReturnValue(undefined)
+    const mockWaitForTx = jest.spyOn(txMonitor, 'waitForTx')
+    const mockWaitForRelayedTx = jest.spyOn(txMonitor, 'waitForRelayedTx')
+
+    renderHook(() => useTxMonitor())
+
+    expect(mockWaitForTx).not.toHaveBeenCalled()
+    expect(mockWaitForRelayedTx).not.toHaveBeenCalled()
+  })
+
+  it('should not monitor transactions if there are no pending transactions', () => {
+    const mockWaitForTx = jest.spyOn(txMonitor, 'waitForTx')
+    const mockWaitForRelayedTx = jest.spyOn(txMonitor, 'waitForRelayedTx')
+
+    renderHook(() => useTxMonitor, { initialReduxState: { pendingTxs: {} } })
+
+    expect(mockWaitForTx).not.toHaveBeenCalled()
+    expect(mockWaitForRelayedTx).not.toHaveBeenCalled()
+  })
+
+  it('should monitor processing transactions', () => {
+    const mockWaitForTx = jest.spyOn(txMonitor, 'waitForTx')
+    const mockWaitForRelayedTx = jest.spyOn(txMonitor, 'waitForRelayedTx')
+
+    const pendingTx: PendingTxsState = {
+      '123': {
+        chainId: '11155111',
+        safeAddress: '0x123',
```

Comment:
Let's avoid using wrong addresses that are not valid but instead generate legit data.

```suggestion
        safeAddress: faker.finance.ethereumAddress(),
```